pyvelox/pyvelox.h

```diff
@@ -168,4 +348,52 @@ inline void addVeloxBindings(pybind11::module& m, bool asLocalModule = true) {
   rowType.def("names", &RowType::names, "Return the names of the columns");
 }
 
+inline void addVectorBindings(
+    pybind11::module& m,
+    bool asModuleLocalDefinitions = true) {
+  namespace py = pybind11;
+  using namespace facebook::velox;
+  std::shared_ptr<memory::MemoryPool> pool = memory::getDefaultMemoryPool();
+  py::class_<BaseVector, std::shared_ptr<BaseVector>>(
+      m, "BaseVector", py::module_local(asModuleLocalDefinitions))
+      .def_static(
+          "from_list",
+          [pool](const py::list& list) mutable {
+            return pyListToVector(list, pool.get());
+          })
+      .def(
+          "__str__",
+          [](std::shared_ptr<BaseVector>& v) { return v->toString(); })
+      .def("__len__", &BaseVector::size)
+      .def(
+          "__getitem__",
+          [](std::shared_ptr<BaseVector>& v, int64_t idx) {
+            return getItemFromVector(v, idx);
+          })
+      .def(
+          "__setitem__",
```

Comment:
Did we still want `size` if we already have `__len__`? In python to get the length of something you usually do `len(x)`

Reply:
Yes len will do fine ! though size() might help primary c++ devs like me who might forget about len :) .

Reply:
It seems that `type()` is already taken in Python returning the Python class of the object. I'll make `type()` be exposed as `dtype()`. This is also how numpy does it, so this would probably help to be consistent within Python.

Reply:
I've added the remaining functions